### DISCUSSION TITLE
Refine mesh within a subdomain

### DISCUSSION THREAD

**Discussion by patte103** [idaholab/moose#21272] [Q&A Meshing] [answered]
Hello!
I am trying to refine my mesh within a subdomain while the rest of the mesh remains coarse. I am currently using the mesh generators within moose and have been referencing the #19047 discussion. From the documentation, the RefineMeshGenerator seemed to only apply to CartesianMeshGenerator, and whenever I coupled it with my SubdomainBoundingBoxGenerator it resulted in this error: 
```
Your MeshGenerator tree contains multiple possible generator outputs :
"subdomain and one or more of the following from an independent set: "refine"

This may be due to a missing dependency or may be intentional. Please select the final MeshGenerator in
the [Mesh] block with the "final_generator" parameter or add additional dependencies to remove the ambiguity.

```
Is there a way to refine my mesh in the subdomain specified region only? Or, which combination of mesh generators will yield the ideal result?

My current Mesh coupled with RefineMeshGenerator:
```
[Mesh]
[gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 80
    ny = 40
    xmin = 0.0
    ymin = 0.0
    xmax = 8000.0
    ymax = 4000.0
    elem_type = QUAD4
  []
  [./subdomain]
    type = SubdomainBoundingBoxGenerator
    input = gen
    bottom_left = '0 1500 0'
    top_right = '8000 2500 0'
    block_id = 1
   [../]
 [refine]
   type = RefineBlockGenerator
   input = gen
    block = '1'
    refinement = '4'
    show_info = true
  []
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

Your tree of MGs is wrong. The inputs to these two cannot both be 'gen'. They need to be chained. 
Maybe try input = subdomain for the refine MG

Guillaume

**Reply by patte103** [reply]
That worked! Thanks so much Guillaume!
